horovod/run/common/util/safe_shell_exec.py

```diff
@@ -134,8 +138,20 @@ def kill_executor_children_if_sigterm_received():
     stdout_fwd.start()
     stderr_fwd.start()
 
+    def kill_middleman_if_master_thread_terminate():
+        event.wait()
+        os.kill(middleman_pid, signal.SIGTERM)
+
+    # TODO: Currently this requires explicitly declaration of the event and signal handler to set
+    #  the event (gloo_run.py:151). Need to figure out a generalized way to hide this behind
```

Comment:
Referencing a specific line will probably go out of date quickly, better to reference the file and possibly function name.